File: code/richediting/src/main/java/org/fbme/ide/richediting/inspections/ECCInspectionsData.java
package org.fbme.ide.richediting.inspections;

import org.fbme.lib.iec61499.ecc.StateTransition;
import org.fbme.scenes.controllers.scene.SceneStateKey;

import java.awt.*;
import java.util.HashMap;

public class ECCInspectionsData {

    public static final SceneStateKey<ECCInspectionsData> KEY = new SceneStateKey<>("ecc-inspections");

    /*package*/ final HashMap<StateTransition, Color> highlightedTransitions = new HashMap<>();
}

File: code/richediting/src/main/kotlin/org/fbme/ide/richediting/inspections/ECCInspectionsData.kt
package org.fbme.ide.richediting.inspections

import org.fbme.lib.iec61499.ecc.StateTransition
import org.fbme.scenes.controllers.scene.SceneStateKey
import java.awt.Color

class ECCInspectionsData {
    /*package*/
    @JvmField
    val highlightedTransitions = HashMap<StateTransition, Color?>()

    companion object {
        @JvmField
        val KEY = SceneStateKey<ECCInspectionsData>("ecc-inspections")
    }
}
